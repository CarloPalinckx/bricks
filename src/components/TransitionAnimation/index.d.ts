import { SFC } from 'react';
import { StyledPropsType } from './style';
declare type PropsType = {
    show: boolean;
    animation: StyledPropsType['animation'];
    stayMounted?: boolean;
};
declare const TransitionAnimation: SFC<PropsType>;
export default TransitionAnimation;
export { PropsType };
