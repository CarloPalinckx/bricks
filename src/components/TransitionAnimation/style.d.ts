import { default as _R } from 'react';
import { StyledComponentClass as _S } from 'styled-components';
import _T from '../../types/ThemeType';
declare type AnimationStateType = 'entering' | 'entered' | 'exiting' | 'exited';
declare type StyledPropsType = {
    animation: 'zoom' | 'fade' | 'slideUp' | 'slideDown' | 'slideLeft' | 'slideRight';
    state: AnimationStateType;
};
declare const StyledAnimation: _S<_R.ClassAttributes<HTMLDivElement> & _R.HTMLAttributes<HTMLDivElement> & StyledPropsType, _T, _R.ClassAttributes<HTMLDivElement> & _R.HTMLAttributes<HTMLDivElement> & StyledPropsType>;
export default StyledAnimation;
export { AnimationStateType, StyledPropsType };
