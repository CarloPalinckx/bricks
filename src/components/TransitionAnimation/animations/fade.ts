import { AnimationStateType } from '../style';

const fade = (state: AnimationStateType): string => {
    switch (state) {
        case 'exiting':
            return `
            @keyframes fadeOut {
                0% { opacity: 1; }
                100% { opacity: 0; }
            }

            animation: fadeOut .3s both;
        `;
        case 'exited':
            return `
            opacity: 0;
        `;
        default:
            return `
            @keyframes fadeIn {
                0% { opacity: 0; }
                100% { opacity: 1; }
            }

            animation: fadeIn .3s both;
        `;
    }
};

export default fade;
