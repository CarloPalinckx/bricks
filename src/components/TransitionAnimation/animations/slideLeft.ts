import { AnimationStateType } from '../style';

const slide = (state: AnimationStateType): string => {
    switch (state) {
        case 'exiting':
            return `
            @keyframes slideOut {
                0% {
                    transform: translate3d(0, 0, 0);
                }
                100% {
                    visibility: hidden;
                    transform: translate3d(-100%, 0, 0);
                }
            }

            animation: slideOut .3s both;
        `;
        case 'exited':
            return `
            transform: translate3d(0, 0, 0);
        `;
        default:
            return `
            @keyframes slideIn {
                0% {
                    transform: translate3d(-100%, 0, 0);
                }
                100% {
                    transform: translate3d(0, 0, 0);
                }
            }

            animation: slideIn .3s both;
        `;
    }
};

export default slide;
