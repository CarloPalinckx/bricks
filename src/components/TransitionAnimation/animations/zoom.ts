import { AnimationStateType } from '../style';

const zoom = (state: AnimationStateType): string => {
    switch (state) {
        case 'exiting':
            return `
            @keyframes zoomOut {
                0% { opacity: 1; transform: scale3d(1, 1, 1); }
                100% { opacity: 0; transform: scale3d(.5, .5, .5); }
            }

            animation: zoomOut .3s both;
        `;
        case 'exited':
            return `
            opacity: 0;
            transform: scale3d(.5, .5, .5);
        `;
        default:
            return `
            @keyframes zoomIn {
                0% { opacity: 0; transform: scale3d(.5, .5, .5); }
                100% { opacity: 1; transform: scale3d(1, 1, 1); }
            }

            animation: zoomIn .3s both;
        `;
    }
};

export default zoom;
