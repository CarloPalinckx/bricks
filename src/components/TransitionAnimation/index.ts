export { default } from './TransitionAnimation.template';
