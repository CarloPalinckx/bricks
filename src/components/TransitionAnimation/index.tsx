import React, { FunctionComponent } from 'react';
import { Transition } from 'react-transition-group';
import StyledAnimation, { StyledPropsType } from './style';

type PropsType = {
    show: boolean;
    animation: StyledPropsType['animation'];
    stayMounted?: boolean;
};

const TransitionAnimation: FunctionComponent<PropsType> = (props): JSX.Element => {
    const unmount = props.stayMounted !== undefined ? !props.stayMounted : true;

    return (
        <Transition in={props.show} timeout={300} mountOnEnter={unmount} unmountOnExit={unmount}>
            {(state: StyledPropsType['state']): JSX.Element => (
                <StyledAnimation animation={props.animation} state={state}>
                    {props.children}
                </StyledAnimation>
            )}
        </Transition>
    );
};

export default TransitionAnimation;
export { PropsType };
