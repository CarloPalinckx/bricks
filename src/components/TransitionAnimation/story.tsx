import { boolean, select } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';
import TransitionAnimation, { PropsType } from '.';
import styled from '../../utility/styled';
import Box from '../Box';
import Heading from '../Heading';

const StyledDiv = styled.div`
    border: solid 3px rgba(255, 36, 94, 0.3);
    padding: 6px 24px;
    border-radius: 5px;
    text-align: center;
`;

storiesOf('TransitionAnimation', module).add('Default', () => {
    const show = boolean('show', true);

    return (
        <Box direction="column">
            <TransitionAnimation
                show={show}
                animation={
                    select(
                        'animation',
                        ['zoom', 'fade', 'slideDown', 'slideUp', 'slideLeft', 'slideRight'],
                        'zoom',
                    ) as PropsType['animation']
                }
            >
                <StyledDiv>
                    <Heading>Lorem ipsum</Heading>
                </StyledDiv>
            </TransitionAnimation>
        </Box>
    );
});
