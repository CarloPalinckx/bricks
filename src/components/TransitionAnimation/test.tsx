import { mount, shallow } from 'enzyme';
import React from 'react';
import { Transition } from 'react-transition-group';
import TransitionAnimation from '.';
import fade from './animations/fade';
import zoom from './animations/zoom';
import slideUp from './animations/slideUp';
import slideDown from './animations/slideDown';
import slideLeft from './animations/slideLeft';
import slideRight from './animations/slideRight';

describe('TransitionAnimation', () => {
    it('should use the fade animation', () => {
        const componentIn = shallow(
            <TransitionAnimation show animation={'fade'}>
                <div>children</div>
            </TransitionAnimation>,
        );

        /* tslint:disable */
        (expect(componentIn.dive()) as any).toHaveStyleRule('animation', 'fadeIn .3s both');
        /* tslint:enable */
    });

    it('should use the slide up animation', () => {
        const componentIn = shallow(
            <TransitionAnimation show animation="slideUp">
                <div>children</div>
            </TransitionAnimation>,
        );

        /* tslint:disable */
        (expect(componentIn.dive()) as any).toHaveStyleRule('animation', 'slideIn .3s both');
        /* tslint:enable */
    });

    it('should use the slide down animation', () => {
        const componentIn = shallow(
            <TransitionAnimation show animation="slideDown">
                <div>children</div>
            </TransitionAnimation>,
        );

        /* tslint:disable */
        (expect(componentIn.dive()) as any).toHaveStyleRule('animation', 'slideIn .3s both');
        /* tslint:enable */
    });

    it('should use the slide left animation', () => {
        const componentIn = shallow(
            <TransitionAnimation show animation="slideLeft">
                <div>children</div>
            </TransitionAnimation>,
        );

        /* tslint:disable */
        (expect(componentIn.dive()) as any).toHaveStyleRule('animation', 'slideIn .3s both');
        /* tslint:enable */
    });

    it('should use the slide right animation', () => {
        const componentIn = shallow(
            <TransitionAnimation show animation="slideRight">
                <div>children</div>
            </TransitionAnimation>,
        );

        /* tslint:disable */
        (expect(componentIn.dive()) as any).toHaveStyleRule('animation', 'slideIn .3s both');
        /* tslint:enable */
    });

    it('should stay mounted', () => {
        const component = mount(
            <TransitionAnimation show={false} animation="zoom" stayMounted>
                <div>children</div>
            </TransitionAnimation>,
        );

        expect(component.find(Transition).prop('mountOnEnter')).toBe(false);
        expect(component.find(Transition).prop('unmountOnExit')).toBe(false);
    });
});

describe('TransitionAnimation/animations', () => {
    it('should zoom in/out', () => {
        expect(zoom('entering')).toMatchSnapshot();
        expect(zoom('exiting')).toMatchSnapshot();
        expect(zoom('exited')).toMatchSnapshot();
    });

    it('should fade in/out', () => {
        expect(fade('entering')).toMatchSnapshot();
        expect(fade('exiting')).toMatchSnapshot();
        expect(fade('exited')).toMatchSnapshot();
    });

    it('should slide up', () => {
        expect(slideUp('entering')).toMatchSnapshot();
        expect(slideUp('exiting')).toMatchSnapshot();
        expect(slideUp('exited')).toMatchSnapshot();
    });

    it('should slide down', () => {
        expect(slideDown('entering')).toMatchSnapshot();
        expect(slideDown('exiting')).toMatchSnapshot();
        expect(slideDown('exited')).toMatchSnapshot();
    });

    it('should slide left', () => {
        expect(slideLeft('entering')).toMatchSnapshot();
        expect(slideLeft('exiting')).toMatchSnapshot();
        expect(slideLeft('exited')).toMatchSnapshot();
    });

    it('should slide right', () => {
        expect(slideRight('entering')).toMatchSnapshot();
        expect(slideRight('exiting')).toMatchSnapshot();
        expect(slideRight('exited')).toMatchSnapshot();
    });
});
