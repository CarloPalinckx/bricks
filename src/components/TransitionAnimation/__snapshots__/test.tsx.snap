// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransitionAnimation/animations should fade in/out 1`] = `
"
            @keyframes fadeIn {
                0% { opacity: 0; }
                100% { opacity: 1; }
            }

            animation: fadeIn .3s both;
        "
`;

exports[`TransitionAnimation/animations should fade in/out 2`] = `
"
            @keyframes fadeOut {
                0% { opacity: 1; }
                100% { opacity: 0; }
            }

            animation: fadeOut .3s both;
        "
`;

exports[`TransitionAnimation/animations should fade in/out 3`] = `
"
            opacity: 0;
        "
`;

exports[`TransitionAnimation/animations should slide down 1`] = `
"
            @keyframes slideIn {
                0% {
                    transform: translate3d(0, -100%, 0);
                }
                100% {
                    transform: translate3d(0, 0, 0);
                }
            }

            animation: slideIn .3s both;
        "
`;

exports[`TransitionAnimation/animations should slide down 2`] = `
"
            @keyframes slideOut {
                0% {
                    transform: translate3d(0, 0, 0);
                }
                100% {
                    visibility: hidden;
                    transform: translate3d(0, -100%, 0);
                }
            }

            animation: slideOut .3s both;
        "
`;

exports[`TransitionAnimation/animations should slide down 3`] = `
"
            transform: translate3d(0, 0, 0);
        "
`;

exports[`TransitionAnimation/animations should slide left 1`] = `
"
            @keyframes slideIn {
                0% {
                    transform: translate3d(-100%, 0, 0);
                }
                100% {
                    transform: translate3d(0, 0, 0);
                }
            }

            animation: slideIn .3s both;
        "
`;

exports[`TransitionAnimation/animations should slide left 2`] = `
"
            @keyframes slideOut {
                0% {
                    transform: translate3d(0, 0, 0);
                }
                100% {
                    visibility: hidden;
                    transform: translate3d(-100%, 0, 0);
                }
            }

            animation: slideOut .3s both;
        "
`;

exports[`TransitionAnimation/animations should slide left 3`] = `
"
            transform: translate3d(0, 0, 0);
        "
`;

exports[`TransitionAnimation/animations should slide right 1`] = `
"
            @keyframes slideIn {
                0% {
                    transform: translate3d(100%, 0, 0);
                }
                100% {
                    transform: translate3d(0, 0, 0);
                }
            }

            animation: slideIn .3s both;
        "
`;

exports[`TransitionAnimation/animations should slide right 2`] = `
"
            @keyframes slideOut {
                0% {
                    transform: translate3d(0, 0, 0);
                }
                100% {
                    visibility: hidden;
                    transform: translate3d(100%, 0, 0);
                }
            }

            animation: slideOut .3s both;
        "
`;

exports[`TransitionAnimation/animations should slide right 3`] = `
"
            transform: translate3d(0, 0, 0);
        "
`;

exports[`TransitionAnimation/animations should slide up 1`] = `
"
            @keyframes slideIn {
                0% {
                    transform: translate3d(0, 100%, 0);
                }
                100% {
                    transform: translate3d(0, 0, 0);
                }
            }

            animation: slideIn .3s both;
        "
`;

exports[`TransitionAnimation/animations should slide up 2`] = `
"
            @keyframes slideOut {
                0% {
                    transform: translate3d(0, 0, 0);
                }
                100% {
                    transform: translate3d(0, 100%, 0);
                    visibility: hidden;
                }
            }

            animation: slideOut .3s both;
        "
`;

exports[`TransitionAnimation/animations should slide up 3`] = `
"
            transform: translate3d(0, 0, 0);
        "
`;

exports[`TransitionAnimation/animations should zoom in/out 1`] = `
"
            @keyframes zoomIn {
                0% { opacity: 0; transform: scale3d(.5, .5, .5); }
                100% { opacity: 1; transform: scale3d(1, 1, 1); }
            }

            animation: zoomIn .3s both;
        "
`;

exports[`TransitionAnimation/animations should zoom in/out 2`] = `
"
            @keyframes zoomOut {
                0% { opacity: 1; transform: scale3d(1, 1, 1); }
                100% { opacity: 0; transform: scale3d(.5, .5, .5); }
            }

            animation: zoomOut .3s both;
        "
`;

exports[`TransitionAnimation/animations should zoom in/out 3`] = `
"
            opacity: 0;
            transform: scale3d(.5, .5, .5);
        "
`;
